packages/block-editor/src/components/inserter-draggable-blocks/index.js

```diff
@@ -23,6 +24,9 @@ const InserterDraggableBlocks = ( {
 		<Draggable
 			__experimentalTransferDataType="wp-blocks"
 			transferData={ transferData }
+			onDragStart={ ( event ) => {
+				event.dataTransfer.setData( 'text/html', serialize( blocks ) );
```

Comment:
This is the thing that I don't like with this approach. In other words, this is just another `transferData` prop. So maybe a better way to go about this is to update the `Draggable` component to offer an API to set multiple data transfer and data types instead of doing it adhoc in a callback.

Reply:
I don't think it's _adhoc_ as it is a web standard api. We should expect components consumers to do anything with `event` in the `onDragStart` event callback, otherwise we probably shouldn't need to expose `event` in the first place?

I get what you mean though, and I'm open to ideas to polish this component if needed! What do you have in mind for such an API? I have to admit that I'm a bit hesitant to introduce too many APIs to do the same thing, while we can just expose standardized web API so that we don't have to worry too much about compact and doc.

The first thing that came to my mind is allowing `transferData` to be an array of types, but that will be a breaking change. So, adding a new prop to indicate that it's an array? Or maybe a `transferDataList` prop that can't coexist with `transferData`? 🤔

Reply:
> while we can just expose standardized web API so that we don't have to worry too much about compact and doc.

Why have a Draggable component in the first place by this logic?

> Or maybe a transferDataList prop that can't coexist with transferDataList? 🤔

That sounds good to me.

Reply:
> Why have a Draggable component in the first place by this logic?

I thought it was because we want to abstract away the custom drag image logic when dragging a block? My point is not that we can't use abstracted API or utility function, but not to over-abstract away standard API that's not too low-level or complicated to use (IMHO).

In this case, users have to look up what exactly `transferDataList` is and how that is different than the original `transferData` prop, while they could just call `event.dataTransfer.setData()` if they're already familiar with the standard API.

FWIW, users still have to call `event.dataTransfer.getData()` anyway in the drop event. A pair of matching APIs looks more intuitive to me.

Reply:
> FWIW, users still have to call event.dataTransfer.getData() anyway in the drop event. A pair of matching APIs looks more intuitive to me.

Yeah, I saw that recently and I think that's a bad design in the drop hook. Ideally users should say I consume this type of data, just give me the data in my callback.

Reply:
> Ideally users should say I consume this type of data, just give me the data in my callback.

`event.dataTransfer.getData()` reads exactly like that to me 😆 .

Anyway, what do you prefer? Do we add an inconsistent prop API for `<Draggable>` and maybe change how drop events are handled in other follow up PRs, or do we just use `event.dataTransfer.getData()` as is in this PR to make the scope smaller?

I still prefer a simpler standard API and I think introducing yet another prop is a mistake. But that's just my opinion and there's no right answer :). And I want to get this in more so I'm willing to do anything 😆.

Reply:
Personally I don't see why having a "dataTransfer" and "__experimentalTransferDataType" and at the same time "onDragStart" to do the exact same thing (set data transfer) is a good thing. 

I also don't see why we have draggable and drop zone abstractions if we don't abstract anything and just use the standard API. 

I'm fine with the current PR assuming the delay change in `onDragStart` is not a breaking change (hard to tell). But I don't think it's ideal to mix abstracted props with standard API.